utils/interfaces/_agent.py

```diff
@@ -141,3 +141,27 @@ def get_spans_list(self, request):
 
     def get_dsm_data(self):
         return self.get_data(path_filters="/api/v0.1/pipeline_stats")
+
+    def get_stats(self, resource=""):
+        """Attempts to fetch the stats the agent will submit to the backend.
+
+        When a valid request is given, then we filter the stats to the ones sampled
+        during that request's execution, and only return those.
+        """
+        logger.debug("WE GETTIN THE STATS")
+
+        for data in self.get_data(path_filters="/api/v0.2/stats"):
+            logger.debug(f"WHATS THE DATA {data}")
+            if "Stats" not in data["request"]["content"]:
```

Comment:
This should be enforced in schemas, rather than here. Ping me if you need any help.

Reply:
I think I figured it out / added it in the right place but let me know if I should make any changes! I didn't add every single possible field here, but this seemed like a good start?

Reply:
That's perfect. Those schema are a very convenient way to ensure that all libs send the valid payloads, up to you to choose the level of rigidity you want to put here :)